lib/ui/painting/multi_frame_codec.cc

```diff
@@ -122,29 +123,40 @@ sk_sp<SkImage> MultiFrameCodec::State::GetNextFrameImage(
     lastRequiredFrame_ = std::make_unique<SkBitmap>(bitmap);
     lastRequiredFrameIndex_ = nextFrameIndex_;
   }
-
-  if (resourceContext) {
-    SkPixmap pixmap(bitmap.info(), bitmap.pixelRef()->pixels(),
-                    bitmap.pixelRef()->rowBytes());
-    return SkImage::MakeCrossContextFromPixmap(resourceContext.get(), pixmap,
-                                               true);
-  } else {
-    // Defer decoding until time of draw later on the raster thread. Can happen
-    // when GL operations are currently forbidden such as in the background
-    // on iOS.
-    return SkImage::MakeFromBitmap(bitmap);
-  }
+  sk_sp<SkImage> result;
+
+  gpu_disable_sync_switch->Execute(
+      fml::SyncSwitch::Handlers()
+          .SetIfTrue([&result, &bitmap] {
+            // Defer decoding until time of draw later on the raster thread. Can
+            // happen when GL operations are currently forbidden such as in the
+            // background on iOS.
```

Comment:
This comment goes with the other MakeFromBitmap call.

Reply:
Done